### PR TITLE
Milestone 5: Interactions

### PR DESCRIPTION
**Pull request by HeflerDev**
Releasing the 5th milestone, which deals with click handlers and (some) basic CSS to ease the evaluation process.
In this release, the **buttons work** accordingly, doing Math operations, handling edge-cases with all tests still passing.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+    /* eslint-enable react/no-unused-state */
+    this.handleClick = this.handleClick.bind(this);
+  }
 
-export default App;
+  handleClick(buttonName) {
+    this.setState(state => (
+      calculate(state, buttonName)
+    ));
+  }
```
Comment (SpaYco) [inline src/components/App.js]: - [x] `handleClick` should be inside the constructor

```js
  constructor(props) {
    super(props);
    /* eslint-disable react/no-unused-state */
    this.state = {
      total: '0',
      next: '',
      operation: '',
    };
    /* eslint-enable react/no-unused-state */

    this.handleClick = buttonName => {
      this.setState(state => (
        calculate(state, buttonName)
      ));
    };
  }
```

File: src/components/ButtonPanel.js [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+  return (
+    <>
+      <div className="button-container">
+        {
+          Object.keys(groupButtons).map(k => (
+            <div className="button-row" key={k}>
+              {
+                groupButtons[`${k}`].map(item => (
+                  <Button key={item} name={item} clickHandler={handleClick2} />
+                ))
```
Comment (SpaYco) [inline src/components/ButtonPanel.js]: - [x] keep `handleClick2` named as `handleClick` without the number

File: src/components/Button.js [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+const Button = ({ name, clickHandler }) => {
+  const handleClick3 = btnName => clickHandler(btnName);
+
+  return (
+    <>
+      <button
+        className="calculator-button"
+        type="button"
+        onClick={() => handleClick3(name)}
+        tabIndex={name}
```
Comment (SpaYco) [inline src/components/Button.js]: - [x] you don't need to make a function here, simply pass `clickHandler` directly to the button.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [x] check below for required changes 👇 

- [x] `OPTIONAL` since this isn't from this milestone, this would be optional after I click the equal button, and I get the total, then I click another key, the display should reset and show me only the number I clicked.

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by HeflerDev** [comment]
- [x] Implemented Required Changes
- [x] Implemented the Optional Change

**Review by uche-inyama — APPROVED** [review]
## Approved


### Hi @HeflerDev :wave: We are good to go :100: :walking: 
 

### Have a great day :slightly_smiling_face: 

#### ``` Uchechukwu Henry Inyama ```
